### PR TITLE
Rename EEType to MethodTable

### PR DESCRIPTION
**Pull request by jkotas**
MethodTable is very established name in CoreCLR. Having different name for the runtime type structure makes it hard to share code with CoreCLR.

For now, I have kept names of other related types (EETypePtr, EETypeCreator, compiler node types, etc.) intact.

### CODE REVIEW

**Review by MichalStrehovsky — APPROVED** [review]
😭 it was a good run.

_To this day_ whenever I see `MethodTable` my brain thinks "a method" and it takes active thinking to convince my brain I'm actually looking at a type. Even Mono's Klass is a better name.

Maybe now I can hammer my brain into thinking the way a VM developer with years spent in the VM thinks, where this name makes any sense.

**Comment by jkotas** [comment]
Yeah, I will miss the EEType name too.

File: src/coreclr/nativeaot/System.Private.TypeLoader/src/Internal/TypeSystem/RuntimeNoMetadataType.cs [hunk 1]
```diff
@@ -139,26 +139,26 @@ protected override TypeFlags ComputeTypeFlags(TypeFlags mask)
             {
                 unsafe
                 {
-                    EEType* eetype = _genericTypeDefinition.ToEETypePtr();
-                    EETypeElementType elementType = eetype->ElementType;
+                    MethodTable* MethodTable = _genericTypeDefinition.ToEETypePtr();
```
Comment (kant2002) [inline src/coreclr/nativeaot/System.Private.TypeLoader/src/Internal/TypeSystem/RuntimeNoMetadataType.cs:142]: should not this be `methodTable` ?

File: src/coreclr/tools/Common/TypeSystem/IL/CoreRTILProvider.cs [hunk 2]
```diff
@@ -82,7 +82,7 @@ private MethodIL TryGetIntrinsicMethodIL(MethodDesc method)
                             return GetCanonTypeIntrinsic.EmitIL(method);
                     }
                     break;
-                case "EEType":
+                case "MethodTable":
```
Comment (kant2002) [inline src/coreclr/tools/Common/TypeSystem/IL/CoreRTILProvider.cs:85]: Does this mean that any other custom runtime will break if `bool SupportsRelativePointers` declared in source code?

File: src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/DependencyAnalysis/ObjectWriter.cs [hunk 3]
```diff
@@ -1135,9 +1135,9 @@ public static void EmitObject(string objectFilePath, IEnumerable<DependencyNode>
                         objectWriter.EmitDebugFunctionInfo(node, nodeContents.Data.Length);
                     }
 
-                    if (node is ConstructedEETypeNode eeType)
+                    if (node is ConstructedEETypeNode MethodTable)
```
Comment (kant2002) [inline src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/DependencyAnalysis/ObjectWriter.cs:1138]: Lower case `methodTable' ?

**Comment by kant2002** [comment]
I was trying to understand how this affects anybody who was going route of custom runtime? I assume not, since only loose related place which I was see was `MethodTable.SupportsRelativePointers` and unlikely somebody implement that.

File: src/coreclr/tools/Common/TypeSystem/IL/CoreRTILProvider.cs (hunk 2 above)
Comment (jkotas) [reply src/coreclr/tools/Common/TypeSystem/IL/CoreRTILProvider.cs:85]: Yep

File: src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/DependencyAnalysis/ObjectWriter.cs (hunk 3 above)
Comment (jkotas) [reply src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/DependencyAnalysis/ObjectWriter.cs:1138]:
```suggestion
                    if (node is ConstructedEETypeNode eeType)
```

File: src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/DependencyAnalysis/ObjectWriter.cs [hunk 4]
```diff
@@ -1135,9 +1135,9 @@ public static void EmitObject(string objectFilePath, IEnumerable<DependencyNode>
...
                     {
-                        objectWriter._userDefinedTypeDescriptor.GetTypeIndex(eeType.Type, needsCompleteType: true);
+                        objectWriter._userDefinedTypeDescriptor.GetTypeIndex(MethodTable.Type, needsCompleteType: true);
```
Comment (jkotas) [inline src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/DependencyAnalysis/ObjectWriter.cs:1140]:
```suggestion
                        objectWriter._userDefinedTypeDescriptor.GetTypeIndex(eeType, needsCompleteType: true);
```

File: src/coreclr/nativeaot/System.Private.TypeLoader/src/Internal/TypeSystem/RuntimeNoMetadataType.cs (hunk 1 above)
Comment (jkotas) [reply src/coreclr/nativeaot/System.Private.TypeLoader/src/Internal/TypeSystem/RuntimeNoMetadataType.cs:142]:
```suggestion
                    MethodTable* pEEType = _genericTypeDefinition.ToEETypePtr();
```

File: src/coreclr/nativeaot/System.Private.TypeLoader/src/Internal/TypeSystem/RuntimeNoMetadataType.cs [hunk 5]
```diff
@@ -139,26 +139,26 @@ protected override TypeFlags ComputeTypeFlags(TypeFlags mask)
...
+                    EETypeElementType elementType = MethodTable->ElementType;
```
Comment (jkotas) [inline src/coreclr/nativeaot/System.Private.TypeLoader/src/Internal/TypeSystem/RuntimeNoMetadataType.cs:143]:
```suggestion
                    EETypeElementType elementType = pEEType->ElementType;
```

File: src/coreclr/nativeaot/System.Private.TypeLoader/src/Internal/TypeSystem/RuntimeNoMetadataType.cs [hunk 6]
```diff
@@ -139,26 +139,26 @@ protected override TypeFlags ComputeTypeFlags(TypeFlags mask)
...
                     if (elementType == EETypeElementType.SystemArray)
                     {
                         // System.Array is a regular class in the type system
                         flags |= TypeFlags.Class;
                     }
                     else if (elementType <= EETypeElementType.Double &&
-                        (eetype->IsGenericTypeDefinition || eetype->BaseType == typeof(System.Enum).TypeHandle.ToEETypePtr()))
+                        (MethodTable->IsGenericTypeDefinition || MethodTable->BaseType == typeof(System.Enum).TypeHandle.ToEETypePtr()))
```
Comment (jkotas) [inline src/coreclr/nativeaot/System.Private.TypeLoader/src/Internal/TypeSystem/RuntimeNoMetadataType.cs:150]:
```suggestion
                        (pEEType->IsGenericTypeDefinition || pEEType->BaseType == typeof(System.Enum).TypeHandle.ToEETypePtr()))
```

File: src/coreclr/nativeaot/System.Private.TypeLoader/src/Internal/TypeSystem/RuntimeNoMetadataType.cs [hunk 7]
```diff
@@ -139,26 +139,26 @@ protected override TypeFlags ComputeTypeFlags(TypeFlags mask)
...
+                        // on generic definition MethodTable)
                         flags |= TypeFlags.Enum;
                     }
                     else
                     {
                         // Paranoid check that we handled enums above
-                        Debug.Assert(eetype->IsGenericTypeDefinition ||
-                            eetype->BaseType != typeof(System.Enum).TypeHandle.ToEETypePtr());
+                        Debug.Assert(MethodTable->IsGenericTypeDefinition ||
+                            MethodTable->BaseType != typeof(System.Enum).TypeHandle.ToEETypePtr());
```
Comment (jkotas) [inline src/coreclr/nativeaot/System.Private.TypeLoader/src/Internal/TypeSystem/RuntimeNoMetadataType.cs:161]:
```suggestion
                        Debug.Assert(pEEType->IsGenericTypeDefinition ||
                            pEEType->BaseType != typeof(System.Enum).TypeHandle.ToEETypePtr());
```

File: src/coreclr/nativeaot/System.Private.TypeLoader/src/Internal/TypeSystem/RuntimeNoMetadataType.cs [hunk 8]
```diff
@@ -180,8 +180,8 @@ protected override TypeFlags ComputeTypeFlags(TypeFlags mask)
 
                 unsafe
                 {
-                    EEType* eetype = _genericTypeDefinition.ToEETypePtr();
-                    if (eetype->IsByRefLike)
+                    MethodTable* MethodTable = _genericTypeDefinition.ToEETypePtr();
+                    if (MethodTable->IsByRefLike)
```
Comment (jkotas) [inline src/coreclr/nativeaot/System.Private.TypeLoader/src/Internal/TypeSystem/RuntimeNoMetadataType.cs:184]:
```suggestion
                    MethodTable* pEEType = _genericTypeDefinition.ToEETypePtr();
                    if (pEEType->IsByRefLike)
```

**Comment by jkotas** [comment]
@kant2002 Thank you for reviewing the change!

**Pull request merged**